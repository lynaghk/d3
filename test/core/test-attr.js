require("./../../lib/env-js/envjs/node");
require("./../../lib/sizzle/sizzle");
require("./../../d3");

var svg = d3.select("body").append("svg:svg")
    .attr("width", "960")
    .attr("height", "500");

console.log("constant attr:");
console.log("  ", document.body.innerHTML);
console.log("");

d3.select("body").selectAll("div")
    .data("abcdefghijk".split(""))
  .enter().append("div")
    .attr("value", function(d, i) { return d; })
    .attr("index", function(d, i) { return i; });

console.log("function attr:");
d3.selectAll("div").each(function() {
  var d = d3.select(this);
  console.log("  ", d.attr("value"), d.attr("index"));
});
console.log("");

d3.select("body").selectAll("div").remove();

svg.attr({
  width: null,
  height: null,
  left: "10",
  top: function() { return 20; }
});

console.log("map attr:");
console.log("  ", document.body.innerHTML);
console.log("");

svg.attr(function() {
  return {
    foo: "foo",
    bar: 42,
    left: null,
    top: null
  };
});

console.log("map function attr:");
console.log("  ", document.body.innerHTML);
console.log("");
