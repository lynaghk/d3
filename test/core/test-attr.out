constant attr:
   <svg width="960" height="500"/>

function attr:
   a 0
   b 1
   c 2
   d 3
   e 4
   f 5
   g 6
   h 7
   i 8
   j 9
   k 10

map attr:
   <svg left="10" top="20"/>

map function attr:
   <svg foo="foo" bar="42"/>

